src/main/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKReadFilterPluginDescriptor.java

```diff
@@ -297,33 +293,29 @@ public CountingReadFilter getMergedCountingReadFilter(final SAMFileHeader samHea
         Utils.nonNull(samHeader);
         return getMergedReadFilter(
                 samHeader,
-                f -> new CountingReadFilter(f),
-                (f1, f2) -> f1.and(f2)
+                CountingReadFilter::fromList
         );
     }
 
     /**
      * Merge the default filters with the users's command line read filter requests, then initialize
      * the resulting filters.
      *
-     * @param samHeader - a SAMFileHeader to initialize read filter instances. May not be null.
-     * @param wrapperFunction - a wrapper function for each read filter; used to wrap read filters in
-     *                        CountingReadFilters
-     * @param mergeFunction - function to use to merge ReadFilters, usually ReadFilter.and
+     * @param samHeader a SAMFileHeader to initialize read filter instances. May not be null.
+     * @param aggregateFunction function to use to merge ReadFilters, usually ReadFilter::fromList. The function
+     *                          must return a default filter of the appropriate type if passed a null list.
      * @param <T> extends ReadFilter, type returned by the wrapperFunction
      * @return Single merged read filter.
      */
     public <T extends ReadFilter> T getMergedReadFilter(
             final SAMFileHeader samHeader,
-            final Function<ReadFilter, T> wrapperFunction,
-            final BinaryOperator<T> mergeFunction) {
+            final BiFunction<List<ReadFilter>, SAMFileHeader, T> aggregateFunction) {
 
         Utils.nonNull(samHeader);
-        Utils.nonNull(wrapperFunction);
-        Utils.nonNull(mergeFunction);
+        Utils.nonNull(aggregateFunction);
 
         if (disableAllReadFilters) {
-            return wrapperFunction.apply(ReadFilterLibrary.ALLOW_ALL_READS);
+            return aggregateFunction.apply(null, samHeader);
```

Comment:
It does with respect to this PR. That is, the two aggregate functions implementations we have return ALLOW_ALL_READS. However, I'll change the javadoc for the aggregate function argument in this method and codify that it should return a wrapped ALLOW_ALL_READS when passed null.